src/Cubie/Moves/Edges.hs

```diff
@@ -0,0 +1,73 @@
+module Cubie.Moves.Edges
+( front
+, back
+, left
+, right
+, up
+, down
+) where
+
+import qualified Cubie.Moves as M (accumulate, permute)
+
+-- set the bound on the accumulator
+accumulate = M.accumulate (2 :: Int) -- TODO: how do i fix type declarations so this cast isnt needed?
+
+{-
+ - Apply a front face rotation to the given orientation vectors
+ -}
+front (edgeOrien, edgePerm) =
+    (ov, pv)
+    where
+        ov = M.permute edgeOrien move
+        pv = M.permute edgePerm move
+        move = [0, 1, 2, 3, 4, 9, 8, 7, 5, 6, 10, 11]
+
+{-
+ - Apply a back face rotation to the given orientation vectors
+ -}
+back (edgeOrien, edgePerm) =
+    (ov, pv)
+    where
+        ov = M.permute edgeOrien move
+        pv = M.permute edgePerm move
+        move = [0, 1, 3, 2, 11, 5, 6, 1,0, 8, 9, 4, 7]
+
+{-
+ - Apply a left face rotation to the given orientation vectors
+ -}
+left (edgeOrien, edgePerm) =
+    (ov, pv)
+    where
+        ov = M.permute edgeOrien move
+        pv = M.permute edgePerm move
+        move = [4, 5, 3, 2, 1, 0, 6, 7, 8, 9, 10, 11]
+
+{-
+ - Apply a right face rotation to the given orientation vectors
+ -}
+right (edgeOrien, edgePerm) =
+    (ov, pv)
+    where
+        ov = M.permute edgeOrien move
+        pv = M.permute edgePerm move
+        move = [0, 1, 7, 6, 4, 5, 2, 3, 8, 9, 10, 11]
+
+{-
+ - Apply a up face rotation to the given orientation vectors
+ -}
+up (edgeOrien, edgePerm) =
+    (ov, pv)
+    where
+        ov = accumulate (M.permute edgeOrien move) $ zip [0, 3, 8, 11] [1, 1..]
+        pv = M.permute edgePerm move
+        move = [8, 1, 2, 11, 4, 5, 6, 7, 3, 9, 10, 0]
```

Comment:
@SaxJ do you have any ideas about how to confirm or test these permutations are correct? I figure adding unit tests for this is probably not going to give much benefit but let me know what you think.  
I got these sequences from the PDF in #1, accounting for the numbering issues for corners 3 and 4. I also went through each to confirm but could have made mistakes